pkg/dynamicconfig/reflectors.go

```diff
@@ -4,54 +4,118 @@
 	}
+	sources, overrides, err := parseConfigs(c)
+	if err != nil {
+		l.Error("Processing configs", logfields.Error, err)
+		return []k8s.ReflectorConfig[DynamicConfig]{}
+	}
+
+	processedSources := map[resolver.ConfigSource]struct{}{} // Ensure no source duplicates
+	var reflectors []k8s.ReflectorConfig[DynamicConfig]
+	sLen := len(sources)
+	for i, source := range sources {
+		if _, ok := processedSources[source]; ok {
+			continue
+		}
+
+		var reflector k8s.ReflectorConfig[DynamicConfig]
+		if source.Kind == resolver.KindConfigMap {
+			reflector = configMapReflector(source.Name, source.Namespace, cs, t, i, sLen, overrides)
+		} else if source.Kind == resolver.KindNodeConfig {
+			reflector = ciliumNodeConfigReflector(source.Name, source.Namespace, cs, t, i, sLen, overrides)
+		} else if source.Kind == resolver.KindNode {
+			reflector = ciliumNodeReflector(source.Name, cs, t, l, i, sLen, overrides)
+		}
+
+		reflectors = append(reflectors, reflector)
+		processedSources[source] = struct{}{}
+	}
+
+	return reflectors
+}
+func parseConfigs(c config) ([]resolver.ConfigSource, resolver.ConfigOverride, error) {
+	var sources []resolver.ConfigSource
+	if err := json.Unmarshal([]byte(c.ConfigSources), &sources); err != nil {
+		return nil, resolver.ConfigOverride{}, fmt.Errorf("error during unmarshall config-sources: %w", err)
+	}
+
+	var overrides resolver.ConfigOverride
+	if err := json.Unmarshal([]byte(c.ConfigSourcesOverrides), &overrides); err != nil {
+		return nil, resolver.ConfigOverride{}, fmt.Errorf("error during unmarshall config-sources: %w", err)
+	}
+	return sources, overrides, nil
+}
+
+func getPriorityForKey(key string, overrides resolver.ConfigOverride, index int, sourceLen int) int {
+	// The first source have no restriction for overrides
+	if index == 0 {
+		return sourceLen
+	}
+
+	// If the len(AllowConfigKeys) > 0, DenyConfigKeys should be ignored
+	if len(overrides.AllowConfigKeys) > 0 {
+		if slices.Contains(overrides.AllowConfigKeys, key) {
+			return sourceLen - index
+		}
+		return sourceLen + index
+	}
 
-	return []k8s.ReflectorConfig[DynamicConfig]{
-		configMapReflector(CiliumConfigMap, cs, t),
+	// If the AllowConfigKeys is empty, check DenyConfigKeys
+	if len(overrides.DenyConfigKeys) > 0 && slices.Contains(overrides.DenyConfigKeys, key) {
+		return sourceLen + index
 	}
+
+	return sourceLen - index
 }
 
-func configMapReflector(name string, cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig]) k8s.ReflectorConfig[DynamicConfig] {
+func configMapReflector(name string, namespace string, cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig], index int, sourceLen int, overrides resolver.ConfigOverride) k8s.ReflectorConfig[DynamicConfig] {
 	return k8s.ReflectorConfig[DynamicConfig]{
-		Name:  "cm-" + name,
+		Name:  "cm-" + name + "-" + namespace,
 		Table: t,
 		TransformMany: func(o any) []DynamicConfig {
 			var entries []DynamicConfig
 			cm := o.(*v1.ConfigMap).DeepCopy()
```

Comment:
Why is the deep copy needed here?

Reply:
Due to the way reflectors were implemented, o is an interface part of https://github.com/cilium/cilium/blob/d1031ce573d76a10e858bfca95505316a16e030b/pkg/k8s/statedb.go#L232h. To prevent modification to the object processed by TransformMany.